### PR TITLE
Filter available_apps by name

### PR DESCRIPTION
**Pull request by cosmastech**
To resolve https://github.com/cogent3/cogent3/issues/1633

- [x] Confirm that test case passes

### CODE REVIEW

File: tests/test_app/test_init.py [hunk 1]
```diff
@@ -70,6 +70,15 @@ def test_available_apps(self):
         self.assertIsInstance(apps, Table)
         self.assertTrue(apps.shape[0] > 10)
 
+    def test_available_apps_filter(self):
```
Comment (GavinHuttley) [inline tests/test_app/test_init.py:73]: we're migrating to all new tests being written as pytest functions.

This test can be converted by just moving the test method to being a test function at the bottom of the file, removing the `self` etc...

Also, we should modify the test so it exactly check the values in the output. So here's my suggested replacement for your test

```python
def test_available_apps_filterl()
    '''"availabe_apps can be filtered by name'''''
    app_name_filter: str = "load"
    apps = available_apps (app_name_filter)
    assert isinstance(apps, Table)
    # check every returned table row 'name' has query is in the
    assert len(apps) > 0
    assert sum(app_name_filter in n for n in t.columns["name"]) == len(table)
```

**Review by GavinHuttley — CHANGES_REQUESTED** [review]
Thanks so much for your contribution! Can you fix the small logic error and the test.

File: tests/test_app/test_init.py (hunk 1 above)
Comment (cosmastech) [reply tests/test_app/test_init.py]: I'm very new to Python. What's the reason for the switch to pytest from unittest?

Comment (GavinHuttley) [reply tests/test_app/test_init.py]: There are many reasons, but the 2 main reasons. `pytest` provides mechanisms that simplify:

1. checking combinations of parameters for a test, a small amount of code (`pytest.mark.parametrize`)
2. elegant test setup and teardown (`pytest.fixture`)

**Review by GavinHuttley — APPROVED** [review]
Awesome job, thanks for the contribution!

**Comment by cosmastech** [comment]
> Awesome job, thanks for the contribution!

Thanks for your patience with a newbie.

**Pull request merged**